src/Controller/SponsorshipLevelBenefit/Edit.php

```diff
@@ -0,0 +1,115 @@
+        CsrfTokenManagerInterface $csrfTokenManager
+    )
+    {
+        $this->renderer = $renderer;
+        $this->sponsorshipLevelBenefitManager = $sponsorshipLevelBenefitManager;
+        $this->csrfTokenManager = $csrfTokenManager;
+    }
+
+    /**
+     * @param FormatSponsorshipLevelBenefit $formatSponsorshipLevelBenefit
+     * @param SponsorshipLevelManagerInterface $sponsorshipLevelManager
+     * @return Response
+     * @throws \Twig_Error_Loader
+     * @throws \Twig_Error_Runtime
+     * @throws \Twig_Error_Syntax
+     */
+    public function handle(FormatSponsorshipLevelBenefit $formatSponsorshipLevelBenefit, SponsorshipLevelManagerInterface $sponsorshipLevelManager): Response
+    {
+        return new Response($this->renderer->render('sponsorshipLevelBenefit/edit.html.twig', [
+            'levels' => $sponsorshipLevelManager->getOrderedList(),
+            'levelBenefits' => $formatSponsorshipLevelBenefit->format(),
+        ]));
+    }
+
+    public function editSponsorshipLevelPosition(Request $request, SponsorshipLevelManagerInterface $sponsorshipLevelManager): JsonResponse
+    {
+        $token = new CsrfToken('sponsorship-level-benefit', $request->request->get('_token'));
+        if (!$this->csrfTokenManager->isTokenValid($token)) {
+            return new JsonResponse([false]);
+        }
+        
+        $move = $request->request->get('move');
+        $id = $request->request->get('id');
+
+        $new_level = $sponsorshipLevelManager->switchPosition($move, $id);
+
+        return new JsonResponse(['new_level_id' => $new_level->getId()]);
+    }
+
+    public function editSponsorshipBenefitPosition(Request $request, SponsorshipBenefitManagerInterface $sponsorshipBenefitManager): JsonResponse
+    {
+        $token = new CsrfToken('sponsorship-level-benefit', $request->request->get('_token'));
+        if (!$this->csrfTokenManager->isTokenValid($token)) {
+            return new JsonResponse([false]);
+        }
+
+        $move = $request->request->get('move');
+        $id = $request->request->get('id');
+
+        $new_benefit = $sponsorshipBenefitManager->switchPosition($move, $id);
+
+        return new JsonResponse(['new_benefit_id' => $new_benefit->getId()]);
+    }
+
+    public function editDatas(Request $request, SponsorshipBenefitManagerInterface $sponsorshipBenefitManager, SponsorshipLevelManagerInterface $sponsorshipLevelManager): JsonResponse
+    {
+        $token = new CsrfToken('sponsorship-level-benefit', $request->request->get('_token'));
+        if (!$this->csrfTokenManager->isTokenValid($token)) {
+            return new JsonResponse([false]);
+        }
+
+        $checked = $request->request->get('is_checked') == "false" ? false : true;
+        $text = $request->request->get('text') ? $request->request->get('text') : null;
+        $sponsorshipBenefit = $sponsorshipBenefitManager->getRepository()->find($request->request->get('benefit_id'));
```

Comment:
You should not leak your repository from the manager. Please create a proxy method at least in the manager that encapsulate the call to the repository.
Thinking Archi Hexa, you should have your repo in Domain/Infra and your Manager in App/Domain, and as such you should never leak the implementation or concept in your UI